app/scripts/directives/oscSourceSecrets.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+
+angular.module("openshiftConsole")
+
+  .directive("oscSourceSecrets", function($uibModal, SecretsService) {
+    return {
+      restrict: 'E',
+      scope: {
+        pickedSecrets: "=model",
+        secretsByType: '=',
+        strategyType: '=',
+        type: "@",
+        displayType: "@",
+        namespace: "=",
+        alerts: '=',
+        serviceAccountToLink: '@'
+      },
+      templateUrl: 'views/directives/osc-source-secrets.html',
+      controller: function($scope) {
+
+        $scope.canAddSourceSecret = function() {
+          var lastSecret = _.last($scope.pickedSecrets);
+          return ($scope.strategyType === "Custom") ? (!_.isEmpty(lastSecret.secretSource.name) && !_.isEmpty(lastSecret.mountPath)) : (!_.isEmpty(lastSecret.secret.name) && !_.isEmpty(lastSecret.destinationDir));
```

Comment:
Shouldn't need `_.isEmpty` in these checks?

```js
lastSecret.secretSource.name && ...
```

I find it too hard to read with everything on one line.

```js
switch ($scope.strategyType) {
case 'Custom':
  return lastSecret.secretSource.name && lastSecret.mountPath;
default:
  return lastSecret.name && lastSecret.destinationDir;
}
```